src/azure-cli/azure/cli/command_modules/cosmosdb/_validators.py

```diff
@@ -19,6 +21,19 @@ def validate_ip_range_filter(ns):
         ns.ip_range_filter = ",".join(ns.ip_range_filter)
 
 
+def validate_private_endpoint_connection_id(ns):
+    if ns.connection_id:
+        id_parts = ns.connection_id.split('/')
+        ns.private_endpoint_connection_name = id_parts[-1]
+        ns.account_name = id_parts[-3]
+        ns.resource_group_name = id_parts[-7]
+
+    if not all([ns.account_name, ns.resource_group_name, ns.private_endpoint_connection_name]):
+        raise CLIError(None, 'incorrect usage: [--id ID | --name NAME --account-name NAME]')
```

Comment:
cosmos db is not unique within one subscription. I think it should be `--name NAME --account-name NAME -g NAME`

Reply:
Thanks for the catch